cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -288,19 +284,13 @@ where
             Ok(yaml) => match serde_yaml::from_slice(&yaml) {
                 Ok(value) => t_vec.push(value),
                 Err(e) => {
-                    let mut old_block: Option<graph_data_model::BlockedEdge> = None;
-                    if is_conditional_edge {
-                        // conditional-edge file threw error while parsing the file but matched
-                        // blocked-edge, so not committing error. It'll be considered as blocked-edge
-                        // till we're able to parse conditional-edge file without errors.
-                        // we'll throw an error if the file is not able to match blocked-edge
-                        // as well as conditional-edge
-                        old_block = match serde_yaml::from_slice(&yaml) {
-                            Ok(value) => value,
-                            Err(_e) => None,
-                        };
-                    }
-                    if old_block.is_none() {
+                    // conditional-edge file threw error while parsing but matched blocked-edge or
+                    // blocked-edge file threw an error and matched conditional edge, dont throw a warning, these are expected
+                    // and doesnt need warning as at the end of the process, it'll match atleast one block.
+                    // if its neither blocked edge nor conditional edge, throw a warning to make sure we log this as an issue.
+                    if !(TypeId::of::<T>() == TypeId::of::<graph_data_model::ConditionalEdgeYaml>())
+                        && !(TypeId::of::<T>() == TypeId::of::<graph_data_model::BlockedEdge>())
+                    {
```

Comment:
We still need to check `T`, because we use this method to also parse channels in https://github.com/openshift/cincinnati/blob/58d65c7a7e242d2d3e7820965afc501d0e955270/cincinnati/src/plugins/internal/graph_builder/openshift_secondary_metadata_parser/plugin.rs#L587 and only "parses as" check would override errors if somehow a blocked edge ends in channel directory.

I think we need a check like this:

if 
(T is BlockedEdge and yaml does not parse as ConditionalEdge)
or
(T is ConditionalEdge and yaml does not parse as Blocked)
then error